### PR TITLE
feature: support hybrid linear attention in sglang integration (contiguous layout)

### PR DESCRIPTION
**Pull request by ztang2370**
## Summary
Adds kvcached support for SGLang hybrid-linear models (attention + mamba). `ElasticHybridLinearKVPool` routes the outer pool's allocator through kvcached; `ElasticMambaPool` backs the inner mamba state with on-demand kvcached VM.

## Scope
Contiguous kvcached layout only (mamba state stays a single (num_mamba_layers, size+1, *shape) tensor per kind). `ElasticMambaPool` raises if `KVCACHED_CONTIGUOUS_LAYOUT=false`; non-contiguous can follow. Speculative decoding `intermediate_*` buffers remain eager — different slot count = different lifecycle.

## Changes

- `integration/sglang/patches.py`:

  - `ElasticHybridLinearKVPoolPatch`: subclass of `HybridLinearKVPool` with a kvcached_allocator property delegating to the inner `full_kv_pool` so `ElasticPagedTokenToKVPoolAllocator` can reach the manager via `kvcache.kvcached_allocator`.

  - `ElasticMambaPoolPatch`: subclass of `MambaPool` that skips the parent's eager `torch.zeros` and routes `alloc/free/clear/available_size` through a dedicated KVCacheManager (block_size=1, num_kv_buffers=1, reserve_null_block=True covers the padded slot 0).

- `integration/sglang/interfaces.py`: new `alloc_mamba_states` packs conv states and temporal state end-to-end per slot in a super-cell, allocates a single contiguous FTensor via `create_kv_tensors(num_kv_buffers=1)`, and `as_strided`s per-state views. Pads `cell_size` up to the smallest divisor of `PAGE_SIZE ≥ raw_cell_size` so `blocks_per_page` is integer; otherwise PageAllocator under-delivers and the scheduler hits "Not enough space for mamba cache" prematurely.

- `integration/sglang/autopatch.py`: registers `ElasticMambaPoolPatch` before `ElasticHybridLinearKVPoolPatch` so `MambaPool` is aliased before `HybridLinearKVPool.__init__` constructs one.

## Test plan
- [x] Hybrid-linear model with `KVCACHED_CONTIGUOUS_LAYOUT=true`

### CODE REVIEW

File: kvcached/integration/sglang/patches.py [hunk 1]
```diff
@@ -689,6 +689,350 @@ def alias_mla_mem_pool_to_elastic(self, mem_pool_mod: types.ModuleType) -> bool:
...
+            class ElasticHybridLinearKVPool(HybridLinearKVPool):  # type: ignore[misc, valid-type]
+                """Hybrid linear-attention KV pool backed by kvcached for
+                the full-attention layers.
+
+                The inner ``full_kv_pool`` is an elastic MHA/MLA pool (because
+                ``MHATokenToKVPool`` / ``MLATokenToKVPool`` are aliased), so the
+                base ``HybridLinearKVPool.__init__`` already wires kvcached for
+                the attention layers. ``MambaPool`` remains a plain torch tensor
+                allocation, managed independently of kvcached. #FIXME
+                """
```
[the remark this answers is not in the document]
Comment (ztang2370) [reply kvcached/integration/sglang/patches.py]: updated

File: kvcached/integration/sglang/patches.py [hunk 2]
```diff
@@ -689,6 +689,350 @@ def alias_mla_mem_pool_to_elastic(self, mem_pool_mod: types.ModuleType) -> bool:
...
+
+                @property
+                def kvcached_allocator(self):
+                    # Delegate to the attention pool's kvcached allocator so
+                    # the outer ElasticPagedTokenToKVPoolAllocator can reach it
+                    # through kvcache.kvcached_allocator.
+                    return self.full_kv_pool.kvcached_allocator
```
[the remark this answers is not in the document]
Comment (ztang2370) [reply kvcached/integration/sglang/patches.py:1012]: The current delegation is right. SGLang drives the two pools through two independent allocator paths, not one:

- Attention: `PagedTokenToKVPoolAllocator` → `kvcache.kvcached_allocator` (the property in question) → token IDs into attention tensors.
- Mamba: `HybridReqToTokenPool.alloc(reqs)` calls `self.mamba_pool.alloc(1)` per request directly. Free/clear/available_size all reach mamba via `req_to_token_pool.mamba_pool.*`, never through the token allocator.

The two have different index spaces (N tokens/req vs. 1 slot/req), so they can't share an allocator.

On the two concerns:

"`ElasticMambaPool.alloc()` is never called." It is, because `MambaPool` is aliased to `ElasticMambaPool` by `ElasticMambaPoolPatch`, so every existing `mamba_pool.alloc()` call site lands on the override, which calls mamba's own `kvcached_allocator` and zeros state.
"Mamba's manager won't track allocations." It does, on the mamba path. Mamba has its own `KVCacheManager` with its own `group_id`, driven by `ElasticMambaPool.alloc/free/clear`.
